server/src/main/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -167,6 +168,15 @@ public static long sizeOfRelocatingShards(
 
     @Override
     public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
+        /*
+         Remote shards were introduced as a part of searchable snapshots. The following block
+         enables allocation for remote shards without restrictions since they do not use the local storage as
+         the primary source of data storage.
+         */
+        if (RoutingPool.REMOTE_CAPABLE.equals(RoutingPool.getNodePool(node))
```

Comment:
can we extract the check (if condition) into a method since same logic is used twice in this class in canAllocate and canRemain. a suggestion for naming is `isRemoteStoredShard`